src/native/tab_bar.rs

```diff
@@ -380,68 +419,147 @@ where
+                bounds: Rectangle {
+                    x: icon_bounds.center_x(),
+                    y: icon_bounds.center_y(),
+                    ..icon_bounds
+                },
+                color: style.icon_color,
+                horizontal_alignment: Horizontal::Center,
+                vertical_alignment: Vertical::Center,
+            });
+        }
+        TabLabel::Text(text) => {
+            let text_bounds = label_layout_children
+                .next()
+                .expect("Graphics: Layout should have a text layout for a Text")
+                .bounds();
+
+            renderer.fill_text(iced_native::text::Text {
+                content: text.to_string(),
+                font: text_font,
+                size: text_bounds.height,
+                bounds: Rectangle {
+                    x: text_bounds.center_x(),
+                    y: text_bounds.center_y(),
+                    ..text_bounds
+                },
+                color: style.text_color,
+                horizontal_alignment: Horizontal::Center,
+                vertical_alignment: Vertical::Center,
+            });
+        }
+        TabLabel::IconText(icon, text) => {
+            let icon_bounds = label_layout_children
+                .next()
+                .expect("Graphics: Layout should have an icons layout for an IconText")
+                .bounds();
+            let text_bounds = label_layout_children
+                .next()
+                .expect("Graphics: Layout should have a text layout for an IconText")
+                .bounds();
+
+            renderer.fill_text(iced_native::text::Text {
+                content: icon.to_string(),
+                font: icon_font,
+                size: icon_bounds.height,
+                bounds: Rectangle {
+                    x: icon_bounds.center_x(),
+                    y: icon_bounds.center_y(),
+                    ..icon_bounds
+                },
+                color: style.icon_color,
+                horizontal_alignment: Horizontal::Center,
+                vertical_alignment: Vertical::Center,
+            });
+
+            renderer.fill_text(iced_native::text::Text {
+                content: text.to_string(),
+                font: text_font,
+                size: text_bounds.height,
+                bounds: Rectangle {
+                    x: text_bounds.center_x(),
+                    y: text_bounds.center_y(),
+                    ..text_bounds
+                },
+                color: style.text_color,
+                horizontal_alignment: Horizontal::Center,
+                vertical_alignment: Vertical::Center,
+            });
+        }
+    };
+
+    if let Some(cross_layout) = children.next() {
+        let cross_bounds = cross_layout.bounds();
+        let is_mouse_over_cross = cross_bounds.contains(cursor_position);
+
+        renderer.fill_text(iced_native::text::Text {
+            content: icons::Icon::X.into(),
```

Comment:
We will have to find out for the future what `into` does here as there is no `Into<&str>` implementation for `Icon`. It could be that the rust compiler turns this into a `String` first and then just references it. I don't know yet. Therefore we will leave this for now as it is.